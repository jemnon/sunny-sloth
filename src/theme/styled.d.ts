import 'styled-components';
import {
  Breakpoints,
  Colors,
  ColorMode,
  FocusColors,
  Fonts,
  FontSizes,
  Header,
  HoverColors,
  MaxWidths,
  Spacing,
  ZIndex,
} from '../types/theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    breakpoints: Breakpoints;
    colors: Colors;
    focusColors: FocusColors;
    fonts: Fonts;
    fontSizes: FontSizes;
    header: Header;
    hoverColors: HoverColors;
    maxWidths: MaxWidths;
    spacing: Spacing;
    transition: string;
    zIndex: ZIndex;
  }
}
