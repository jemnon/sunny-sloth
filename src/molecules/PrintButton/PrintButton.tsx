import React, { FC } from 'react';
import styled from 'styled-components';
import Button from '../../atoms/Button';
import { Print } from '../../atoms/Icons';
import { Colors } from '../../types/theme';

type ColorScheme = keyof Pick<Colors, 'orange' | 'nearBlack'>;

interface PrintButtonProps {
  colorScheme?: ColorScheme;
  onClick?: () => void;
  width?: string;
}

const PrintButtonContainer = styled.div<Pick<PrintButtonProps, 'width'>>`
  display: inline-flex;
  justify-content: center;
  align-items: center;

  width: ${({ width = 'auto' }): string => width};
`;
const PrintButtonLabel = styled.span`
  padding-right: ${({ theme }): string => theme.spacing.sm1};
`;

const PrintButton: FC<PrintButtonProps> = ({
  colorScheme = 'orange',
  onClick,
  width,
}) => (
  <PrintButtonContainer width={width}>
    <Button
      colorScheme={colorScheme}
      onClick={onClick}
      variant="outline"
      shape="rectangle"
      size="small"
      width={width}
    >
      <PrintButtonLabel>Print Recipe</PrintButtonLabel>
      <Print
        fontSize="0.75rem"
        fill={colorScheme === 'nearBlack' ? '#111' : '#BB5B34'}
        style={{ marginTop: '2px' }}
      />
    </Button>
  </PrintButtonContainer>
);

export default PrintButton;
