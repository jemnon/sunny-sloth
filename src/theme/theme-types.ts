export interface Breakpoints {
  sm: string;
  md: string;
  lg: string;
  xlg: string;
}

export interface Fonts {
  lato: string;
  latoBold: string;
  latoItalic: string;
  noto: string;
  notoBold: string;
}

export interface Colors {
  cream: string;
  darkTeal: string;
  teal: string;
  lightBrown: string;
  nearBlack: string;
  nearWhite: string;
  darkOrange: string;
  orange: string;
  white: string;
  red: string;
  green: string;
}

export type ColorMode = 'dark' | 'light';

export interface MaxWidths {
  sm: string;
  md: string;
  lg: string;
}

export interface Spacing {
  sm1: string;
  sm2: string;
  sm3: string;
  sm4: string;
  md1: string;
  md2: string;
  md3: string;
  md4: string;
  lg1: string;
  lg2: string;
  lg3: string;
  lg4: string;
}

export interface ZIndex {
  z0: string;
  z1: string;
  z2: string;
  z3: string;
  z4: string;
  z5: string;
  z6: string;
  z999: string;
  z9999: string;
}
