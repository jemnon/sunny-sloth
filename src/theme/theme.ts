const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
