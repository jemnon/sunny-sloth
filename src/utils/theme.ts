import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
