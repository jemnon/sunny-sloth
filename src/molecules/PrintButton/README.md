# PrintButton

### Description
Component description

```js
import PrintButton from '@fenderdigital/ui/FILL IN TYPEPrintButton';
```

### Props
* `propName` - property name and description 
