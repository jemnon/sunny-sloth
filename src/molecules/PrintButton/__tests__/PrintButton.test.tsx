import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import PrintButton from '../PrintButton';
import theme from '../../../theme';

describe('<PrintButton />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <PrintButton />
    </ThemeProvider>
  );
  it('exists', (): void => {
    expect(PrintButton).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});