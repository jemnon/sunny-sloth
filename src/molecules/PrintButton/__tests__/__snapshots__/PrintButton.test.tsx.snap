// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrintButton /> renders defaults correctly 1`] = `
<div
  className="sc-fFubgz eoTpYJ"
>
  <button
    className="sc-bdfBwQ kEQyMa"
    shape="rectangle"
    size="small"
  >
    <span
      className="sc-bkzZxe iWfDzb"
    >
      Print Recipe
    </span>
    <svg
      className="sc-kEjbxe evyxfF"
      fill="#BB5B34"
      fontSize="0.75rem"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "marginTop": "2px",
        }
      }
      viewBox="0 0 870.9 875"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M667.7 169.9a103.12 103.12 0 00-30.1-72.7l-67.1-67.1A102.93 102.93 0 00497.7 0h-241a53.67 53.67 0 00-53.6 53.6v172.2h464.6v-55.9zM53.6 734.7h92.1V548.4h579.6v186.2h92a53.67 53.67 0 0053.6-53.6V341.8a53.67 53.67 0 00-53.6-53.6H53.6A53.67 53.67 0 000 341.8V681a53.56 53.56 0 0053.6 53.7zm84.8-373.2a56.3 56.3 0 11-56.3 56.3 56.29 56.29 0 0156.3-56.3zm533.2 486.8V602H199.1v246.2a26.93 26.93 0 0026.8 26.8h418.8a26.8 26.8 0 0026.9-26.7zM319.2 711.7h232.3a26.66 26.66 0 0126.5 22.8 24.66 24.66 0 01.3 3.9 26.74 26.74 0 01-26.8 26.8H319.2a26.74 26.74 0 01-26.8-26.8 25.61 25.61 0 01.3-3.9 26.75 26.75 0 0126.5-22.8z"
      />
    </svg>
  </button>
</div>
`;
